RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripRawToClusterAlgo.dev.cc

```diff
@@ -0,0 +1,1096 @@
+
+    // Get the number of strips to unpack
+    nStrips_h_ = cms::alpakatools::make_host_buffer<uint32_t>(queue);
+    const uint32_t size = stripMapping_d_->view().metadata().size();
+    auto viewSrc = make_device_view(queue, stripMapping_d_->view().fedChStripsN(size - 1));
+    auto viewDst = make_host_view(*nStrips_h_->data());
+    alpaka::memcpy(queue, viewDst, viewSrc);
+  }
+
+  void SiStripRawToClusterAlgo::unpackStrips(Queue& queue, const Data* conditions_Data) {
+    // std::cout << "#kerStrips," << nStrips_h->data()[0] << std::endl;
+    // std::cout << "#invalidFed," << invalidFedChN_d.data()[0] << std::endl;
+
+    // Allocate the SiStripDigi collection on device
+    const uint32_t nStrips = nStrips_h_->data()[0];
+    digis_d_ = std::make_unique<SiStripDigiDevice>(nStrips, queue);
+
+    // Run the unpacking kernel
+    uint32_t divider = 32u;
+    uint32_t nChannels = (*stripMapping_d_)->metadata().size();
+    uint32_t groups = divide_up_by(nChannels, divider);
+    auto workDiv = make_workdiv<Acc1D>(groups, divider);
+    alpaka::exec<Acc1D>(queue,
+                        workDiv,
+                        SiStripKer_unpackZS2{},
+                        fedBuffer_d_->data(),
+                        digis_d_->view(),
+                        stripMapping_d_->const_view(),
+                        conditions_Data);
+
+    // dumpUnpackedStrips(queue, digis_d_.get()); // (for debugging)
+
+    // Allocate and initialize the StripClustersAux collection
+    sClustersAux_d_ = std::make_unique<StripClustersAuxDevice>(nStrips, queue);
+    // LogDebug("sClustersAux") << "Size of StripClustersAuxDevice (bytes): " << alpaka::getExtentProduct(sClustersAux_d_->buffer()) * sizeof(std::byte);
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(1u, 1u),
+                        SiStripKer_init{},
+                        channelThreshold_,
+                        seedThreshold_,
+                        clusterThresholdSquared_,
+                        maxSequentialHoles_,
+                        maxSequentialBad_,
+                        maxAdjacentBad_,
+                        minGoodCharge_,
+                        maxClusterSize_,
+                        sClustersAux_d_->view());
+
+    // Cluster seeding
+    alpaka::exec<Acc1D>(queue,
+                        workDiv,
+                        SiStripKer_setSeedStrips{},
+                        digis_d_->const_view(),
+                        sClustersAux_d_->view(),
+                        stripMapping_d_->const_view(),
+                        conditions_Data);
+
+    // Un-seed any contiguous strips in the same detector
+    alpaka::exec<Acc1D>(queue,
+                        workDiv,
+                        SiStripKer_setNCSeedStrips{},
+                        digis_d_->const_view(),
+                        sClustersAux_d_->view(),
+                        stripMapping_d_->const_view());
```

Comment:
In general, very small kernels are not very efficient.

Could you merge `SiStripKer_setSeedStrips` and `SiStripKer_setNCSeedStrips` into a single one ?